opentelemetry-api/src/opentelemetry/trace/__init__.py

```diff
@@ -409,9 +410,80 @@ def use_span(
         yield
 
 
+class ProxyTracer(Tracer):
+    """Proxies all calls to current TracerProvider
+    """
+
+    def __init__(
+        self, get_current_tracer: typing.Callable[[], "Tracer"],
+    ) -> None:
+        self._get_current_tracer = get_current_tracer
+
+    def start_span(
+        self,
+        name: str,
+        context: typing.Optional[Context] = None,
+        kind: SpanKind = SpanKind.INTERNAL,
+        attributes: types.Attributes = None,
+        links: typing.Sequence[Link] = (),
+        start_time: typing.Optional[int] = None,
+        record_exception: bool = True,
+        set_status_on_exception: bool = True,
+    ) -> "Span":
+        return self._get_current_tracer().start_span(
+            name=name,
+            context=context,
+            kind=kind,
+            attributes=attributes,
+            links=links,
+            start_time=start_time,
+            record_exception=record_exception,
+            set_status_on_exception=set_status_on_exception,
+        )
+
+    @contextmanager  # type: ignore
+    def start_as_current_span(
+        self,
+        name: str,
+        context: typing.Optional[Context] = None,
+        kind: SpanKind = SpanKind.INTERNAL,
+        attributes: types.Attributes = None,
+        links: typing.Sequence[Link] = (),
+        start_time: typing.Optional[int] = None,
+        record_exception: bool = True,
+        set_status_on_exception: bool = True,
+    ) -> typing.Iterator["Span"]:
+        with self._get_current_tracer().start_as_current_span(
+            name=name,
+            context=context,
+            kind=kind,
+            attributes=attributes,
+            links=links,
+            start_time=start_time,
+            record_exception=record_exception,
+            set_status_on_exception=set_status_on_exception,
+        ) as span:
+            yield span
+
+    @contextmanager  # type: ignore
+    def use_span(
+        self, span: "Span", end_on_exit: bool = False,
+    ) -> typing.Iterator[None]:
+        with self._get_current_tracer().use_span(
+            span=span, end_on_exit=end_on_exit,
+        ) as context:
+            yield context
+
+
 _TRACER_PROVIDER = None
 
 
+@functools.lru_cache(maxsize=128)  # type: ignore
```

Comment:
Some linter was complaining about not defined `maxsize`. But it could be defined as `None` to make it unbounded — that's make sense here, I'll change.